### PR TITLE
fix: limit=0 results (#5735)

### PR DESCRIPTION
**Pull request by robhafner**
- [] Do only one thing
- [] Non breaking API changes
- [] Tested

### CODE REVIEW

**Comment by robhafner** [comment]
Here is the link for the pull request to fix the sqlite module.

https://github.com/go-gorm/sqlite/pull/114

**Comment by robhafner** [comment]
Here is the link for the pull request to fix the sqlserver module.

https://github.com/go-gorm/sqlserver/pull/70

**Comment by robhafner** [comment]
Please let me know if there is anything else that you need from me to move this pull request forward as it is preventing some of our services from upgrading to gorm 2. 

Also, given that these fixes reside in different modules it's unclear to move as to how to make the build pass.

**Comment by System-Glitch** [comment]
This PR contains breaking changes.

File: clause/limit.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import "strconv"
 
 // Limit limit clause
 type Limit struct {
-	Limit  int
```
Comment (System-Glitch) [inline clause/limit.go:7]: This is a breaking change.

Comment (frankli0324) [reply clause/limit.go:7]: then why is this released? I don't see the point where this change is necessary for the fix...

File: clause/limit_test.go [hunk 2]
```diff
@@ -8,18 +8,26 @@ import (
 )
 
 func TestLimit(t *testing.T) {
+	limit0 := 0
```
Comment (frankli0324) [inline clause/limit_test.go:11]: test cases should not be changed unless for a reason that's good enough, this is not good enough.

Comment (li-jin-gou) [reply clause/limit_test.go:11]: +1

**Pull request merged**